### DISCUSSION TITLE
A way to use already defined tags/categories

### DISCUSSION THREAD

**Discussion by josineto** [StaticJsCMS/static-cms#845] [Ideas]
Hi, I'd like to know whether it's possible to configure a widget to use fields already used in the site. The most obvious example is with categories and tags. For instance, a list of tags for a post in which each list item has a dropdown with all existing tags in the site.

It seems to me a bit linked to the relation widget. A more difficult way to achieve that would be to create a collection for tags (to keep the same example) and retrieve that collection in a relation widget. But this way it would be necessary to create a new collection item each time a new tag is defined, what would be not practical.

*To explain it in details, keeping the same example:*

- I create a new post.
- When I have to type the post tags, the (desired) widget **shows me all site tags**.
- I then choose between these and/or to define new tag(s).
- If I define new tags, these are shown in the widget the next time I create a post.

I think the best way to it (from UX POV) would be a delimiter separated list (as [implemented in 2.4.0](https://github.com/StaticJsCMS/static-cms/releases/tag/v2.4.0)) that suggests already defined values in a dropdown *as I type*. But I don't know whether that is practical or even possible from a developer POV...

**Comment by KaneFreeman** [comment]
This is related to: https://github.com/StaticJsCMS/static-cms/issues/295
